### DISCUSSION TITLE
Why can't assets be uploaded to the database?

### DISCUSSION THREAD

**Discussion by Bilge** [craftcms/cms#10140] [Ideas] [answered]
According to the documentation, assets can either live on the local filesystem, or a cloud provider that specializes specifically in general purpose object storage. Yet, the existing database can also store binary objects and is already available in any valid installation. Why isn't the database a valid storage option?

**Comment by brandonkelly** [comment, marked as answer]
It’s technically possible, and someone has even written a Flysystem adapter for storing files in the DB: https://packagist.org/packages/phlib/flysystem-pdo

Databases are an efficient way to store files, which is why we have not added this to the core. It would be fine if a plugin wanted to add that capability though.
